packages/lit-html/src/lib/lit-html.ts

```diff
@@ -213,33 +222,35 @@ class Template {
     const commentRegex = /-->/g;
     // Comments not started with <!--, like </{, can be ended by a single `>`
     const comment2Regex = />/g;
-    const tagRegex = /(>)|(?:(?:[ \x09\x0a\x0c\x0d])([^\0-\x1F\x7F-\x9F "'>=/]+)([ \x09\x0a\x0c\x0d]*=[ \x09\x0a\x0c\x0d]*([^ \x09\x0a\x0c\x0d"'`<>=]|("|')|$)))/g;
+    const tagRegex = new RegExp(TAG_STATE_REGEX, 'g');
     const singleQuoteAttr = /'/g;
     const doubleQuoteAttr = /"/g;
 
     let html = type === SVG_RESULT ? '<svg>' : '';
-    let quote: string | undefined;
     let node: Node | null;
     let nodeIndex = 0;
     let bindingIndex = 0;
+    let attrNameIndex = 0;
 
     // The current parsing state, represented as a reference to one of the
     // regexes
     let regex = textRegex;
 
-    for (let i = 0; i < strings.length - 1; i++) {
+    for (let i = 0; i < l; i++) {
       const s = strings[i];
       // The index of the end of the last attribute. When this is !== -1 at
       // end of a string, it means we're in a quoted attribute position.
       let attrNameEnd = -1;
+      let attrName: string | undefined = undefined;
       let lastIndex = 0;
+      let match: RegExpExecArray | null;
```

Comment:
Nit: Why are we hoisting this out?

Reply:
Very small size savings by being in this declarators list, especially since this was the only variable declaration in the loop.

Reply:
I'd suggest commenting anything non-obvious micro-opts like this, otherwise they're bound to be "fixed" by someone in the future.